x-pack/plugins/search_indices/public/hooks/api/use_delete_index.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AcknowledgedResponseBase } from '@elastic/elasticsearch/lib/api/types';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useKibana } from '../use_kibana';
+import { useIndex } from './use_index';
+
+export const useDeleteIndex = (indexName: string) => {
+  const { http } = useKibana().services;
+  const indices = [indexName];
+  const body = JSON.stringify({
+    indices,
+  });
+  const queryClient = useQueryClient();
+  const { queryKey } = useIndex(indexName);
```

Comment:
We should probably have a constants for the query keys and likely a function for building the queryKey based on the indexName. 

I think using this hook here will do a fetch of of the index, with the polling, by including this hook here.

Reply:
For an example I experimented with using `enum` for query keys in [search homepage](https://github.com/TattdCodeMonkey/kibana/blob/ca5d165e37a62a875d84ab1a7d1592f60b0fad9e/x-pack/plugins/search_homepage/public/constants.ts) [That PR](https://github.com/elastic/kibana/pull/188491/files) ended up not getting merged for other reasons though.

Reply:
Also move for other UseQuery implementation APIs as well. d7ca01165aa40ac40342ee9a45914fad03e44888